examples/vae/utils/mnist_cached.py

```diff
@@ -218,3 +223,18 @@ def setup_data_loaders(dataset, use_cuda, batch_size, sup_num=None, root='./data
         loaders[mode] = DataLoader(cached_data[mode], batch_size=batch_size, shuffle=True, **kwargs)
 
     return loaders
+
+
+def mkdir_p(path):
+    try:
+        os.makedirs(path)
+    except OSError as exc:  # Python >2.5
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
+
+
+EXAMPLE_DIR = os.path.dirname(os.path.abspath(os.path.join(__file__, os.pardir)))
```

Comment:
Is there an advantage to `os.pardir` over repeated `os.path.dirname(os.path.dirname(-))`?